hvcat doc example fixed

mbeltagy opened the pull request:
hvcat doc example was not working 
I changed it to one that does.

mbeltagy committed:
hvcat doc example fixed 

hvcat doc example was not working 
I changed it to one that does.

tkelman commented:
Note that the docstrings in the file you edited are auto-generated. The same change would also need to be made at the source for the docstring, https://github.com/JuliaLang/julia/blob/6aa15b859e1b114061439fa6640173d2978fed3d/base/docs/helpdb.jl#L3795

dpsanders commented:
Probably the example would be clearer with a 2x3 matrix, as was originally (presumably) intended:
```julia
julia> a, b, c, d, e, f = 1, 2, 3, 4, 5, 6
(1,2,3,4,5,6)

julia> [a b c; d e f]
2x3 Array{Int64,2}:
 1  2  3
 4  5  6
```
This can then be a doctest.

mbeltagy committed:
Added doctest for hvcat 

Removed test for a non-working example and added clearer doctest example instead.

mbeltagy commented:
I fixed that now with a couple of clear doctest examples in my last commit.

tkelman commented on base/docs/helpdb.jl:
```diff
@@ -3792,7 +3792,34 @@ utf8(s)
+2x3 Array{Int64,2}:
+ 1  2  3
+ 4  5  6
+ 
```
please remove the trailing whitespace from here and 3810, and if you've got a local source build it would also be handy if you could run `make && usr/bin/julia doc/genstdlib.jl` on this branch to propagate these changes into the RST.

mbeltagy replied:
I followed your instructions. The latest commit should be good.

tkelman replied:
Looks good, thanks!

mbeltagy committed:
fixing whitespaces and generating arrary.rst doc
